site/app/libraries/database/DatabaseQueries.php

```diff
@@ -1287,10 +1362,18 @@ public function updateGradeableData(Gradeable $gradeable) {
      * @param GradeableComponent $component
      */
     public function insertGradeableComponentData($gd_id, GradeableComponent $component) {
```

Comment:
I need the if statement because if $component->getVerifier() is null, $component->getVerifier()->getId() will throw an error (calling getId() on a null)

Reply:
Ok, I see that, but you can still achieve the same query without duplicating everything.